internal/core/metadata/rest_device.go

```diff
@@ -558,8 +558,7 @@ func restSetDeviceOpStateById(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Check if the device exists
-	var d models.Device
-	err := dbClient.GetDeviceById(&d, did)
+	d, err := dbClient.GetDeviceById(did)
```

Comment:
See, like here the `var` was eliminated. I think we should do this wherever possible.